xla/service/gpu/transforms/pipelined_p2p_rewriter.cc

```diff
@@ -667,6 +667,13 @@ absl::StatusOr<bool> ProcessComputation(
                                      instruction_sequence, hlo));
     changed = true;
   }
+  // After processing all the operations, if nothing set the flag
+  // collective_in_computation[computation] to true, then there must be no
+  // collective in this computation.
+  if (collective_in_computation.find(computation) ==
+      collective_in_computation.end()) {
+    collective_in_computation[computation] = false;
+  }
```

Comment:
Nit: this could just be `collective_in_computation.insert({computation, false});`.  `insert` is a no-op if the key already exists in the `flat_hash_map`.

(If you do make this change please also add a comment that explaining this.)